EasyPost/CreditCard.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using RestSharp;
+
+namespace EasyPost
+{
+    public class CreditCard : Resource
+    {
+        [JsonProperty("id")]
+        public string id { get; set; }
+        [JsonProperty("disabled_at")]
+        public string disabled_at { get; set; }
+        [JsonProperty("object")]
+        public string Object { get; set; }
+        [JsonProperty("name")]
+        public string name { get; set; }
+        [JsonProperty("last4")]
+        public string last4 { get; set; }
+        [JsonProperty("exp_month")]
+        public string exp_month { get; set; }
+        [JsonProperty("exp_year")]
+        public string exp_year { get; set; }
+        [JsonProperty("brand")]
+        public string brand { get; set; }
+
+        /// <summary>
+        ///     Payment method priority
+        /// </summary>
+        public enum Priority
+        {
+            Primary,
+            Secondary
+        }
+
+        /// <summary>
+        ///     Get a payment method (credit card) by priority.
+        /// </summary>
+        /// <param name="priority">Which priority payment method to get.</param>
+        /// <returns>An EasyPost.CreditCard instance.</returns>
+        /// <exception cref="Exception">Billing has not been set up yet, or the Priority provided is invalid.</exception>
+        private static async Task<CreditCard> GetPaymentMethodByPriority(Priority priority)
+        {
+            PaymentMethod paymentMethods = await PaymentMethod.All();
+
+            CreditCard? paymentMethod = null;
+            switch (priority)
+            {
+                case Priority.Primary:
+                    paymentMethod = paymentMethods.primary_payment_method;
+                    break;
+                case Priority.Secondary:
+                    paymentMethod = paymentMethods.secondary_payment_method;
+                    break;
+                default:
+                    break;
+            }
+
+            if (paymentMethod == null || !paymentMethod.id.StartsWith("card_"))
+            {
+                throw new Exception("The chosen payment method is not a valid method. Please try again.");
+            }
+
+            return paymentMethod;
+        }
+
+        /// <summary>
+        ///     Fund a credit card.
+        /// </summary>
+        /// <param name="amount">Amount to fund.</param>
+        /// <param name="paymentMethodId">ID of payment method to fund.</param>
+        /// <returns>An EasyPost.CreditCardFund instance.</returns>
+        private static async Task<CreditCardFund> Fund(string amount, string paymentMethodId)
```

Comment:
I don't feel having two competing interfaces for the same thing is a good idea. There should typically only be one rock solid way of doing something. Since the pattern we chose was for the user to pass an amount and then which of the two payment methods on file they want to use, we should keep that consistent and remove this version of the function that accepts an ID completely.

Reply:
This method is private, so is not available to the end user to call directly.

Reply:
Ah, I missed that.